types/node-forge/index.d.ts

```diff
@@ -107,11 +107,60 @@ declare module "node-forge" {
             extensions: any[];
             publicKey: any;
             md: any;
+            /**
+             * Sets the subject of this certificate.
+             *
+             * @param attrs the array of subject attributes to use.
+             * @param uniqueId an optional a unique ID to use.
+             */
+            setSubject(attrs: any[], uniqueId?: string): void;
+            /**
+              * Sets the subject of this certificate.
+              *
+              * @param attrs the array of subject attributes to use.
+              * @param uniqueId an optional a unique ID to use.
+              */
+            setIssuer(attrs: any[], uniqueId?: string): void;
+            /**
+              * Sets the extensions of this certificate.
+              *
+              * @param exts the array of extensions to use.
+              */
+            setExtensions(exts: any[]): void
```

Comment:
Define and use a `CertificateExtension` type?
I see a bunch of fields here: https://github.com/digitalbazaar/forge/blob/74b7472b2143735fca0c68a7f20ba32c4fbd5397/lib/x509.js#L2086

It looks like you might have to use tagged union types (https://blog.mariusschulz.com/2016/11/03/typescript-2-0-tagged-union-types)

```ts
type CertificateExtension = CertificateExtensionKeyUsage /* | ... */;
interface CertificateExtensionKeyUsage {
  name: 'keyUsage';
  digitalSignature?: string; // type?
  // ...
}
```

Reply:
digitalSignature is number.
'CertificateExtension' is too complicated, I do not have time to add the complete.